llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -111,9 +112,27 @@ void InterBlockScheduling::enterFunction(MachineFunction *MF) {
   // Get ourselves a hazard recognizer
   HR = std::make_unique<AIEHazardRecognizer>(MF->getSubtarget());
 
+  const TargetRegisterInfo *TRI = MF->getSubtarget().getRegisterInfo();
+  LiveRegs MBBLiveness(MF);
+  auto &LiveIns = MBBLiveness.getLiveIns();
+
   // Define our universe of blocks
   for (MachineBasicBlock &MBB : *MF) {
-    Blocks.emplace(&MBB, &MBB);
+    auto Itr = Blocks.emplace(&MBB, &MBB).first;
+    auto &BS = Itr->second;
+    BS.LiveOuts.init(*TRI);
+    // Calculating LiveOuts by iterating over each successor of the MBB and
+    // adding each successor's LiveIns to LiveOuts.
+    for (const MachineBasicBlock *Succ : MBB.successors()) {
+      auto It = LiveIns.find(Succ);
```

Comment:
Please use `const LivePhysRegs &MBBLiveins = LiveIns.at(Succ)` if you don't manually check the iterator.